data/pipeline.py

```diff
@@ -29,4 +27,18 @@ def apply_config_prerpoc(self)->dict:
                 dataset : dict = preproc_fn_dict[key](self.dataset)
             else:
                 raise Exception("No such preprocessing function.")
-        return dataset
\ No newline at end of file
+        
+        self.dataset = dataset
+
+    def save_preproc_dataset(self)->None:
+        """
+        Save the preprocessed dataset.
+        """
+        dump_json_file(self.config["output_path"],self.dataset)
+
+
+
+if __name__ == "__main__":
```

Comment:
Maybe a comment about what this does might be useful? Some sparse documentation.